tests/unittests/config/test_cc_apk_configure.py

```diff
@@ -322,4 +335,82 @@ def test_edge_main_community_testing_local_repos(self):
         self.assertEqual(expected_content, util.load_file(REPO_FILE))
 
 
+class TestApkConfigureSchema:
+    @pytest.mark.parametrize(
+        "config, error_msg",
+        (
+            # Valid schemas
+            ({"apk_repos": {"preserve_repositories": True}}, None),
+            ({"apk_repos": {"alpine_repo": None}}, None),
+            ({"apk_repos": {"alpine_repo": {"version": "v3.21"}}}, None),
+            (
+                {
+                    "apk_repos": {
+                        "alpine_repo": {
+                            "base_url": "http://yep",
+                            "community_enabled": True,
+                            "testing_enabled": True,
+                            "version": "v3.21",
+                        }
+                    }
+                },
+                None,
+            ),
+            ({"apk_repos": {"local_repo_base_url": "http://some"}}, None),
+            # Invalid schemas
+            (
+                {"apk_repos": {"alpine_repo": {"version": False}}},
+                "apk_repos.alpine_repo.version: False is not of type"
+                " 'string'",
+            ),
+            (
+                {
+                    "apk_repos": {
+                        "alpine_repo": {"version": "v3.12", "bogus": 1}
+                    }
+                },
+                "apk_repos.alpine_repo: Additional properties are not allowed"
+                " \('bogus' was unexpected\)",
+            ),
+            (
+                {"apk_repos": {"alpine_repo": {}}},
+                "apk_repos.alpine_repo: 'version' is a required property,"
+                " apk_repos.alpine_repo: {} does not have enough properties",
+            ),
+            (
+                {"apk_repos": {"alpine_repo": True}},
+                "apk_repos.alpine_repo: True is not of type 'object', 'null'",
+            ),
+            (
+                {"apk_repos": {"preserve_repositories": "wrongtype"}},
+                "apk_repos.preserve_repositories: 'wrongtype' is not of type"
+                " 'boolean'",
+            ),
+            (
+                {"apk_repos": {}},
+                "apk_repos: {} does not have enough properties",
+            ),
+            (
+                {"apk_repos": {"local_repo_base_url": None}},
+                "apk_repos.local_repo_base_url: None is not of type 'string'",
+            ),
+        ),
+    )
+    @skipUnlessJsonSchema()
+    @mock.patch("cloudinit.config.schema.read_cfg_paths")
+    def test_schema_validation(self, read_cfg_paths, config, error_msg, paths):
+        read_cfg_paths.return_value = paths
+        # New-style schema $defs exist in config/cloud-init-schema*.json
+        for schema_file in Path(cloud_init_project_dir("config/")).glob(
+            "cloud-init-schema*.json"
+        ):
+            util.copy(schema_file, paths.schema_dir)
+        schema = get_schema()
```

Comment:
We may want to make a json_schema fixture at some point, but I didn't think to create that in this branch. This is a lot of boilerplate setup for us to pull into all 51 test_cc_* modules.